From 43aafa14a21ef6ec10d7066afc150cde8a352fbf Mon Sep 17 00:00:00 2001
From: Roger Quadros <rogerq@ti.com>
Date: Fri, 31 Jan 2014 12:57:55 +0200
Subject: [PATCH 199/285] HACK: usb: musb: core: Prevent Spurious Babble
 interrupt on Resume

Adding a delay in musb_runtime_resume() prevent the spurious
Babble interrupt from happening during a remote wakeup resume.

Fixes D-01343 - AM33XX: CONNECTIVITY: Babble interrupt when using
		keyboard without mouse

Signed-off-by: Roger Quadros <rogerq@ti.com>
[Kevin: The original patch taken from TI
ti-sdk-am437x-evm-07.01.00.00-Linux-x86-Install.bin]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/usb/musb/musb_core.c |    6 ++++++
 1 files changed, 6 insertions(+), 0 deletions(-)

diff --git a/drivers/usb/musb/musb_core.c b/drivers/usb/musb/musb_core.c
index ccac79a..0aded87 100644
--- a/drivers/usb/musb/musb_core.c
+++ b/drivers/usb/musb/musb_core.c
@@ -2322,6 +2322,12 @@ static int musb_resume_noirq(struct device *dev)
 	if (!musb->suspended)
 		return 0;
 
+	/* HACK
+	 * Delay here to prevent Spurious Babble interrupt on
+	 * remote wakeup resume.
+	 */
+	mdelay(20);
+
 	musb_restore_context(musb);
 	musb->suspended = false;
 
-- 
1.7.5.4

